zkvm/src/signature/prover.rs

```diff
@@ -0,0 +1,195 @@
+#![allow(non_snake_case)]
+
+use crate::errors::VMError;
+use crate::signature::multikey::Multikey;
+use crate::signature::musig::*;
+use crate::signature::VerificationKey;
+use crate::transcript::TranscriptProtocol;
+use curve25519_dalek::constants::RISTRETTO_BASEPOINT_POINT;
+use curve25519_dalek::ristretto::RistrettoPoint;
+use curve25519_dalek::scalar::Scalar;
+use merlin::Transcript;
+use rand;
+use subtle::ConstantTimeEq;
+
+#[derive(Copy, Clone, Debug)]
+pub struct Nonce(Scalar);
```

Comment:
i'm not sure we need this type - the nonce is always internal to the signer, so we can simply use Scalar. Same for Privkey type.

Reply:
removed!